noir/noir-repo/acvm-repo/acvm/src/pwg/mod.rs

```diff
@@ -382,6 +406,65 @@ impl<'a, B: BlackBoxFunctionSolver> ACVM<'a, B> {
         }
     }
 
+    fn extract_assertion_payload(
+        &self,
+        location: OpcodeLocation,
+    ) -> Option<ResolvedAssertionPayload> {
+        let found_assertion_payload = self.assertion_payloads.iter().find_map(|(loc, payload)| {
+            if location == *loc {
+                Some(payload)
+            } else {
+                None
+            }
+        });
+        let Some(found_assertion_payload) = found_assertion_payload else {
+            return None;
+        };
+        match found_assertion_payload {
+            AssertionPayload::StaticString(string) => {
+                Some(ResolvedAssertionPayload::String(string.clone()))
+            }
+            AssertionPayload::Raw(error_selector, expression) => {
+                let mut fields = vec![];
+                for expr in expression {
+                    match expr {
+                        ExpressionOrMemory::Expression(expr) => {
+                            let value = get_value(expr, &self.witness_map).ok()?;
+                            fields.push(value);
+                        }
+                        ExpressionOrMemory::Memory(block_id) => {
+                            let memory_block = self.block_solvers.get(block_id)?;
+                            fields.extend((0..memory_block.block_len).map(|memory_index| {
+                                *memory_block
+                                    .block_value
+                                    .get(&memory_index)
+                                    .expect("All memory is initialized on creation")
+                            }));
+                        }
+                    }
+                }
+
+                Some(match *error_selector {
+                    STRING_ERROR_SELECTOR => {
+                        // If the error selector is 0, it means the error is a string
+                        let string = fields
+                            .iter()
+                            .map(|field| {
+                                let as_u8 = field.try_to_u64().unwrap_or_default() as u8;
```

Comment:
Should we `unwrap_or_default()` here? Shouldn't we just panic as that means we have a compiler bug